feat(cast) tx: pretty print new transaction fields

ryanschneider opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests.
-->

## Motivation

I was debugging some blob txs and used `cast tx` to print them, only to learn that `cast` was only outputting the "legacy" fields for transactions.

## Solution

match on the `transaction_type` and output the correct fields.

NOTE: it currently panics w/ a `todo!` for unknown tx types, but we could have it fall back to the `Some(0) | None` case instead.  I figured the panic would make it more likely for the new tx types to get implemented in a timely manner :)

mattsse commented on crates/common/fmt/src/ui.rs:
```diff
@@ -306,10 +306,143 @@ impl UIfmt for OtherFields {
     }
 }
 
+impl UIfmt for AccessListItem {
+    fn pretty(&self) -> String {
+        self.storage_keys.pretty()
+    }
+}
```
this does not print the address?

ryanschneider replied:
Hah yup, adding that now.

mattsse commented on crates/common/fmt/src/ui.rs:
```diff
@@ -306,10 +306,143 @@ impl UIfmt for OtherFields {
-        format!(
-            "
+        match self.transaction_type {
+            Some(3) =>
```
could we order these asc?

ryanschneider replied:
Updated in https://github.com/foundry-rs/foundry/commit/e06f98cc043b4fe2c8a47ac9dafedf272176c866

mattsse commented on crates/common/fmt/src/ui.rs:
```diff
@@ -306,10 +306,143 @@ impl UIfmt for OtherFields {
     fn pretty(&self) -> String {
-        format!(
-            "
+        match self.transaction_type {
```
this is a bit verbose, but I'm okay with this because we want to eventually embed the actual tx object in the rpc response and then we need to do this anyway

mattsse commented on crates/common/fmt/src/ui.rs:
```diff
@@ -306,10 +306,143 @@ impl UIfmt for OtherFields {
-        format!(
-            "
+        match self.transaction_type {
+            Some(3) =>
```
could we also add Some(4) which is the same as 2 but with `authorization_list` field?

ryanschneider replied:
Added in https://github.com/foundry-rs/foundry/commit/e06f98cc043b4fe2c8a47ac9dafedf272176c866

mattsse commented on crates/common/fmt/src/ui.rs:
```diff
@@ -324,22 +457,24 @@ to                   {}
+                    self.value.pretty(),
+                    self.other.pretty()
+                ),
+            _ => todo!("cast needs to be updated to support this new tx type")
```
this would panic for L2 txs like deposit, and we need to handle this variant as well

ryanschneider replied:
Ok, I went with the fallback approach in https://github.com/foundry-rs/foundry/commit/e06f98cc043b4fe2c8a47ac9dafedf272176c866, let me know if there's another way.

mattsse reviewed (changes requested):
supportive, I'm okay with having one format for each variant because not worth it adding some unified impl for this

ryanschneider commented:
@mattsse I think I addressed all of your feedback in https://github.com/foundry-rs/foundry/commit/e06f98cc043b4fe2c8a47ac9dafedf272176c866, PTAL!  Thanks!  I also added unit tests for tx types 1, 2, and 3, but don't know a good source for type 4 txs, has it been enabled on any of the Pectra devnets yet?

ryanschneider commented:
Re-pushed after running `cargo +nightly fmt`

mattsse reviewed (approved):
ty!
